File: ui/hub-mk3-ui/src/store/orm/segment-choice-arrangement-pick.js
// Copyright (c) XJ Music Inc. (https://xj.io) All Rights Reserved.
import {attr, fk, Model} from "redux-orm";

class SegmentChoiceArrangementPick extends Model {
  toString() {
    return `SegmentChoiceArrangementPick: ${this.name}`;
  }

  // Declare any static or instance methods you need.
}

SegmentChoiceArrangementPick.modelName = 'SegmentChoiceArrangementPick';

// Declare your related fields.
SegmentChoiceArrangementPick.fields = {
  id: attr(),
  start: attr(),
  length: attr(),
  amplitude: attr(),
  pitch: attr(),
  name: attr(),
  segmentChoiceArrangementId: fk({
    to: 'SegmentChoiceArrangement',
    as: 'segmentChoiceArrangement',
    relatedName: 'segmentChoiceArrangementPicks',
  }),
  instrumentAudioId: fk({
    to: 'InstrumentAudio',
    as: 'instrumentAudio',
    relatedName: 'segmentChoiceArrangementPicks',
  }),
  programSequencePatternEventId: fk({
    to: 'ProgramSequencePatternEvent',
    as: 'programSequencePatternEvent',
    relatedName: 'segmentChoiceArrangementPicks',
  }),
  segmentId: fk({
    to: 'Segment',
    as: 'segment',
    relatedName: 'segmentChoiceArrangementPicks',
  }),
  programVoiceId: fk({
    to: 'ProgramVoice',
    as: 'programVoice',
    relatedName: 'segmentChoiceArrangementPicks',
  }),
};

export default SegmentChoiceArrangementPick;

File: engine/src/segment/SegmentChoiceArrangementPick.cpp
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.

#include "xjmusic/segment/SegmentChoiceArrangementPick.h"

namespace XJ {

  bool SegmentChoiceArrangementPick::equals(const SegmentChoiceArrangementPick &segmentChoiceArrangementPick) const {
    return id == segmentChoiceArrangementPick.id &&
           segmentId == segmentChoiceArrangementPick.segmentId &&
           segmentChoiceArrangementId == segmentChoiceArrangementPick.segmentChoiceArrangementId &&
           segmentChordVoicingId == segmentChoiceArrangementPick.segmentChordVoicingId &&
           instrumentAudioId == segmentChoiceArrangementPick.instrumentAudioId &&
           programSequencePatternEventId == segmentChoiceArrangementPick.programSequencePatternEventId &&
           startAtSegmentMicros == segmentChoiceArrangementPick.startAtSegmentMicros &&
           lengthMicros == segmentChoiceArrangementPick.lengthMicros &&
           amplitude == segmentChoiceArrangementPick.amplitude &&
           tones == segmentChoiceArrangementPick.tones &&
           event == segmentChoiceArrangementPick.event;
  }

  unsigned long long SegmentChoiceArrangementPick::hashCode() const {
    return std::hash<UUID>{}(id) ^
           std::hash<int>{}(segmentId) ^
           std::hash<UUID>{}(segmentChoiceArrangementId) ^
           std::hash<UUID>{}(segmentChordVoicingId) ^
           std::hash<UUID>{}(instrumentAudioId) ^
           std::hash<UUID>{}(programSequencePatternEventId) ^
           std::hash<long>{}(startAtSegmentMicros) ^
           std::hash<long>{}(lengthMicros) ^
           std::hash<float>{}(amplitude) ^
           std::hash<std::string>{}(tones) ^
           std::hash<std::string>{}(event);
  }

}// namespace XJ
